src/helpers/formulaHelper.js

```diff
@@ -0,0 +1,187 @@
+const formulas = require('./formulas.json')
+
+const TOKEN_TYPE = {
+    Number: Symbol("number"),
+    Variable: Symbol("variable"),
+    Function: Symbol("function"),
+    Operator: Symbol("operator")
+};
+
+const FUNCTION_TOKENS = [
+    "MIN",
+    "MAX",
+    "ROUND"
+];
+
+function getFormulaByKey(key) {
+    return formulas[key];
+}
+
+function parse(formulaObj) {
+    const constants = formulaObj.constants;
+    let tokens = formulaObj.formula.split('');
+
+    let resultStack = [];
+    let parseStack = [];
+
+    for (let i = 0; i < tokens.length; i++) {
+        if (tokens[i] == ' ')
+            continue;
+        
+        if (tokens[i] >= '0' && tokens[i] <= '9') {
+            let number = "";
+            while (i < tokens.length && tokens[i] >= '0' && tokens[i] <= '9') {
+                number += tokens[i];
+                i++;
+            }
+            resultStack.push({ token: parseInt(number), type: TOKEN_TYPE.Number });
+            i--;
+        }
+
+        else if (isValidNonNumberToken(tokens[i])) {
+            let tempToken = "";
+            while (i < tokens.length &&
+                (isValidNonNumberToken(tokens[i]) || (tokens[i] >= '0' && tokens[i] <= '9'))) 
+            {
+                tempToken += tokens[i];
+                i++;
+            }
+            if (constants.hasOwnProperty(tempToken)) {
+                resultStack.push({ token: constants[tempToken], type: TOKEN_TYPE.Number });
+            } else if (FUNCTION_TOKENS.includes(tempToken.toUpperCase())) {
+                parseStack.push(tempToken.toUpperCase());
+            } else {
+                resultStack.push({ token: tempToken, type: TOKEN_TYPE.Variable });
+            }
+
+            i--;
+        }
+
+        else if (tokens[i] == '(' || tokens[i] == ',') {
+            parseStack.push(tokens[i]);
```

Comment:
We should probably extract this into a function even if it's only one line for the sake of consistency